src/participant/prompts/promptBase.ts

```diff
@@ -163,16 +165,27 @@ export abstract class PromptBase<TArgs extends PromptArgsBase> {
   protected getHistoryMessages({
```

Comment:
This function is starting to get a bit long and hard to follow all of the things happening in it. 
It already has the `// eslint-disable-next-line complexity` which is usually an indicator that we should break it into a few functions, even if it comes with a slightly hit to performance (like running through the messages multiple times).
Should we do that now? Break this function into multiple parts where each is doing a certain thing? That'll also make it more easily unit testable.

Reply:
More like a question about this area in code and maybe a bit of a request :) if you will refactor this code, could we split the history into something like `getUserHistoryMessages` and `getAssistantHistoryMessages`. When testing this functionality I found it difficult to print the history content because it is already wrapped into the `vscode.LanguageModelChatMessage` format when it is returned here: https://github.com/mongodb-js/vscode/blob/main/src/participant/prompts/promptBase.ts#L130

Maybe, it could be something like:

```
const messages = [
  vscode.LanguageModelChatMessage.Assistant(this.getAssistantPrompt(args)),
  vscode.LanguageModelChatMessage.Assistant(this.getAssistantHistoryMessages()),
  vscode.LanguageModelChatMessage.User(this.getUserHistoryMessages()),
  vscode.LanguageModelChatMessage.User(prompt),
];
```

Or something like that. The idea here is that we have some unformatted string value to print to see what message we send to the model.

Reply:
I am also dealing changing getHistoryMessages in [VSCODE-632](https://jira.mongodb.org/browse/VSCODE-632) so I think it'll be best to deal with some refactoring for it there.